### PR TITLE
KCE-45-create-tech-expo-challenge

### PR DESCRIPTION
**Pull request by paulagb8894**

### CODE REVIEW

**Review by pablogh-2000 — COMMENTED** [review]
Add Readme with the following content:

### C++ Code challenge

What is missing the in the following code?
Provide a solution that adds the minimal number lines to the code.

```
#include <vector>

class SudokuValidator {
public:
   bool isValid(const std::vector<std::vector<int>>& grid)
   {
       for (int i = 0; i < 9; ++i) {
           std::vector<bool> row_values(9, false);
           std::vector<bool> col_values(9, false);

           for (int j = 0; j < 9; ++j) {
              if (row_values[grid[j][i] - 1] == false) {
                  row_values[grid[j][i] - 1] = true;
              }
              else {
                  return false;
              }
              if (col_values[grid[i][j] - 1] == false) {
                  col_values[grid[i][j] - 1] = true;
              }
              else {
                  return false;
              }
          }
      }

      return true;  
   }
}

```

Drop your solution in paper in the box at Klepsydra's booth or send it electronically to [challenge@klepsydra.com](mailto:challenge@klepsydra.com).

Please provide your name and email address when submitting your solution.

Add website link and logo.

**Review by pablogh-2000 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,49 @@
-# space-tech-challenge-2022
-Space Tech Expo Challange 2022
+<p align="right">
+  <img width="25%" height="25%"src="images/klepsydra_logo.jpg">
+</p>
+
+# C++ Code challenge
+
+## What is missing the in the following code?
+Provide a solution that adds the minimal number lines to the code.
```
Comment (mharshe) [inline README.md:8]: Add this text at the start : A valid Sudoko solution is a 9 × 9 grid with digits such that each column, each row, and each of the nine 3 × 3 subgrids that compose the grid (also called "boxes", "blocks", or "regions") contains all of the digits from 1 to 9. The following code is intended to validate if a solution to Sudoku puzzle is correct or not.

Comment (paulagb8894) [reply README.md:10]: Done

**Review by mharshe — APPROVED** [review]

**Review by pablogh-2000 — APPROVED** [review]

File: include/space_tech_20200.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class SudokuValidator {
+public:
+   bool isValid(const std::vector<std::vector<int>>& grid)
+   {
+       for (int i = 0; i < 9; ++i) {
+           std::vector<bool> rowValues(9, false);
+           std::vector<bool> colValues(9, false);
+
+	   for (int j = 0; j < 9; ++j) {
```
Comment (alfonsoespinosa) [inline include/space_tech_20200.h:12]: This line starts with a tab, and depending on the editor, the line is not shown properly indented. Not very serious issue though.

File: include/space_tech_20200.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+              if (colValues[grid[j][i] - 1] == false) {
+                  colValues[grid[j][i] - 1] = true;
+              }
+              else {
+                  return false;
+              }
+          }
+      }
+      return true;  
+   }
```
Comment (alfonsoespinosa) [inline include/space_tech_20200.h:28]:
```
    bool isValid(const std::vector<std::vector<int>>& grid)
    {
        for (int i = 0; i < 9; ++i) {
            std::vector<bool> rowValues(9, false);
            std::vector<bool> colValues(9, false);

	        for (int j = 0; j < 9; ++j) {
                if (rowValues[grid[i][j] - 1] == false) {
                    rowValues[grid[i][j] - 1] = true;
                }
                else {
                    return false;
                }
                if (colValues[grid[j][i] - 1] == false) {
                    colValues[grid[j][i] - 1] = true;
                }
                else {
                    return false;
                }
            }
        }
        return true;
    }
```

Comment (alfonsoespinosa) [reply include/space_tech_20200.h:28]: Just the indentation issue

Comment (paulagb8894) [reply include/space_tech_20200.h]: Done

**Pull request merged**